### ISSUE TITLE
[Intellij][Spark][PPE] Explorer node refresh error: MismatchedInputException: Cannot construct instance of `java.util.ArrayList` 

### ISSUE THREAD

**Issue by wezhang** [Microsoft/azure-tools-for-java#2697]
Provisioning on PPE, got the following exceptions:

```
2019-01-22 14:26:22,897 [ 159941]   WARN - node.CosmosSparkADLAccountNode - Got exceptions when listing Azure Data Lake account(sextanscaboalki) for listing Spark pools: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: (String)"{"value":[{"errorMessage":"","id":"abf364e6-f8ef-4c58-8b1e-745ccaee6e4a","name":"MyClustertest","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad264.ccsctp.net","analyticsUnits":6,"submitTime":"2019-01-22T06:12:52.373Z","startTime":"2019-01-22T06:12:53.397Z","state":"Scheduled","schedulerState":"Scheduled","result":"None","tags":{},"priority":1000},{"errorMessage":"","id":"4357dced-d6c6-4fbf-be74-06ae8ec2703a","name":"spark-test-cluster","activityType":"spark","entityTyp"[truncated 10431 chars]; line: 1, column: 27] (through reference chain: com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePoolList["value"]->java.util.ArrayList[0]->com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePool["errorMessage"]) 
2019-01-22 14:26:25,430 [ 162474]   INFO - park.CosmosSparkClusterOpsCtrl - Provision message received. AdlAccount: sextanscaboalki, node: sextanscaboalki 
2019-01-22 14:26:26,278 [ 163322]   WARN - rkClusterProvisionCtrlProvider - Can't get the account sextanscaboalki details: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: (String)"{"value":[{"errorMessage":"","id":"abf364e6-f8ef-4c58-8b1e-745ccaee6e4a","name":"MyClustertest","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad264.ccsctp.net","analyticsUnits":6,"submitTime":"2019-01-22T06:12:52.373Z","startTime":"2019-01-22T06:12:53.397Z","state":"Scheduled","schedulerState":"Scheduled","result":"None","tags":{},"priority":1000},{"errorMessage":"","id":"4357dced-d6c6-4fbf-be74-06ae8ec2703a","name":"spark-test-cluster","activityType":"spark","entityTyp"[truncated 10431 chars]; line: 1, column: 27] (through reference chain: com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePoolList["value"]->java.util.ArrayList[0]->com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePool["errorMessage"]) 
2019-01-22 14:26:27,381 [ 164425]   WARN - rkClusterProvisionCtrlProvider - Can't get the account sextanscaboalki details: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: (String)"{"value":[{"errorMessage":"","id":"abf364e6-f8ef-4c58-8b1e-745ccaee6e4a","name":"MyClustertest","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad264.ccsctp.net","analyticsUnits":6,"submitTime":"2019-01-22T06:12:52.373Z","startTime":"2019-01-22T06:12:53.397Z","state":"Scheduled","schedulerState":"Scheduled","result":"None","tags":{},"priority":1000},{"errorMessage":"","id":"4357dced-d6c6-4fbf-be74-06ae8ec2703a","name":"spark-test-cluster","activityType":"spark","entityTyp"[truncated 10431 chars]; line: 1, column: 27] (through reference chain: com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePoolList["value"]->java.util.ArrayList[0]->com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePool["errorMessage"])
	at rx.exceptions.Exceptions.propagate(Exceptions.java:57)
	at com.microsoft.azure.hdinsight.sdk.common.HttpObservable.convertJsonResponseToObject(HttpObservable.java:324)
	at com.microsoft.azure.hdinsight.sdk.common.HttpObservable.lambda$get$7(HttpObservable.java:386)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:69)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:395)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:355)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onNext(OperatorMerge.java:846)
	at rx.observers.Subscribers$5.onNext(Subscribers.java:235)
	at rx.internal.operators.OperatorDoAfterTerminate$1.onNext(OperatorDoAfterTerminate.java:50)
	at rx.internal.util.ScalarSynchronousObservable$WeakSingleProducer.request(ScalarSynchronousObservable.java:276)
	at rx.Subscriber.setProducer(Subscriber.java:211)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at rx.Subscriber.setProducer(Subscriber.java:205)
	at rx.internal.util.ScalarSynchronousObservable$JustOnSubscribe.call(ScalarSynchronousObservable.java:138)
	at rx.internal.util.ScalarSynchronousObservable$JustOnSubscribe.call(ScalarSynchronousObservable.java:129)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeUsing.call(OnSubscribeUsing.java:94)
	at rx.internal.operators.OnSubscribeUsing.call(OnSubscribeUsing.java:32)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:248)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:148)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
	at rx.internal.producers.SingleDelayedProducer.emit(SingleDelayedProducer.java:102)
	at rx.internal.producers.SingleDelayedProducer.setValue(SingleDelayedProducer.java:85)
	at rx.internal.operators.OnSubscribeFromCallable.call(OnSubscribeFromCallable.java:48)
	at rx.internal.operators.OnSubscribeFromCallable.call(OnSubscribeFromCallable.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OnSubscribeSwitchIfEmpty$ParentSubscriber.subscribe(OnSubscribeSwitchIfEmpty.java:104)
	at rx.internal.operators.OnSubscribeSwitchIfEmpty.call(OnSubscribeSwitchIfEmpty.java:52)
	at rx.internal.operators.OnSubscribeSwitchIfEmpty.call(OnSubscribeSwitchIfEmpty.java:31)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:230)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: (String)"{"value":[{"errorMessage":"","id":"abf364e6-f8ef-4c58-8b1e-745ccaee6e4a","name":"MyClustertest","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad264.ccsctp.net","analyticsUnits":6,"submitTime":"2019-01-22T06:12:52.373Z","startTime":"2019-01-22T06:12:53.397Z","state":"Scheduled","schedulerState":"Scheduled","result":"None","tags":{},"priority":1000},{"errorMessage":"","id":"4357dced-d6c6-4fbf-be74-06ae8ec2703a","name":"spark-test-cluster","activityType":"spark","entityTyp"[truncated 10431 chars]; line: 1, column: 27] (through reference chain: com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePoolList["value"]->java.util.ArrayList[0]->com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePool["errorMessage"])
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1342)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1031)
	at com.fasterxml.jackson.databind.deser.ValueInstantiator._createFromStringFallbacks(ValueInstantiator.java:371)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromString(StdValueInstantiator.java:323)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:242)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:136)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:136)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2992)
	at com.microsoft.azure.hdinsight.sdk.rest.ObjectConvertUtils.convertJsonToObject(ObjectConvertUtils.java:46)
	at com.microsoft.azure.hdinsight.sdk.common.HttpObservable.convertJsonResponseToObject(HttpObservable.java:320)
	... 54 more
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.microsoft.azure.hdinsight.sdk.common.HttpResponse.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:118)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:73)
	... 52 more
 
2019-01-22 14:26:44,747 [ 181791]   WARN - node.CosmosSparkADLAccountNode - Got exceptions when listing Azure Data Lake account(sextanscaboalki) for listing Spark pools: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: (String)"{"value":[{"errorMessage":"","id":"b71ea596-8ec4-4cb0-9609-2c9fbf24afe4","name":"zhwe-190122","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad264.ccsctp.net","analyticsUnits":6,"submitTime":"2019-01-22T06:26:43.8109518Z","state":"Submitted","schedulerState":"Submitted","result":"None","tags":{},"priority":1000},{"errorMessage":"","id":"abf364e6-f8ef-4c58-8b1e-745ccaee6e4a","name":"MyClustertest","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad2"[truncated 10760 chars]; line: 1, column: 27] (through reference chain: com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePoolList["value"]->java.util.ArrayList[0]->com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePool["errorMessage"]) 
2019-01-22 14:26:49,942 [ 186986]   WARN - node.CosmosSparkADLAccountNode - Got exceptions when listing Azure Data Lake account(sextanscaboalki) for listing Spark pools: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.ArrayList` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('')
 at [Source: (String)"{"value":[{"errorMessage":"","id":"b71ea596-8ec4-4cb0-9609-2c9fbf24afe4","name":"zhwe-190122","activityType":"spark","entityType":"resourcepools","submitter":"admin@aad264.ccsctp.net","analyticsUnits":6,"submitTime":"2019-01-22T06:26:43.81Z","startTime":"2019-01-22T06:26:44.67Z","state":"Scheduled","schedulerState":"Scheduled","result":"None","tags":{},"priority":1000},{"errorMessage":"","id":"abf364e6-f8ef-4c58-8b1e-745ccaee6e4a","name":"MyClustertest","activityType":"spark","entityType":"resou"[truncated 10793 chars]; line: 1, column: 27] (through reference chain: com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePoolList["value"]->java.util.ArrayList[0]->com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkResourcePool["errorMessage"]) 
```

**Comment by wezhang** [comment]
The `errorMessage` is defined as:
```
    /**
     * The error message details for the activity, if the activity failed.
     */
    @JsonProperty(value = "errorMessage", access = JsonProperty.Access.WRITE_ONLY)
    private List<ErrorDetails> errorMessage;
```

But the response's `errorMessage` value is `""`

**Comment by t-rufang** [comment]
Fixed in #2682

**Comment by v-lanjli** [comment]
Fixed in build 'azure-toolkit-for-intellij-2018.3.develop.979.01-23-2019'. Verified and closed this issue
